shuup/core/tasks.py

```diff
@@ -0,0 +1,97 @@
+# -*- coding: utf-8 -*-
+# This file is part of Shuup.
+#
+# Copyright (c) 2012-2020, Shoop Commerce Ltd. All rights reserved.
+#
+# This source code is licensed under the OSL-3.0 license found in the
+# LICENSE file in the root directory of this source tree.
+import json
+
+from shuup.utils.importing import cached_load, load
+
+
+class TaskNotSerializableError(Exception):
+    """
+    Raised when the task can't be serialized.
+    """
+    pass
+
+
+class Task(object):
+    function = None     # type: str
+    kwargs = None       # type: dict
+
+    def __init__(self, function, **kwargs):
+        """
+        :param function: A string that represents the function specification.
+            It will be locaded dynamically and executed passing the given kwargs.
+            E.g.: `myapp.my_lib.do_domething`
+
+        :param kwargs: Set of parameter to pass to the `function`. The parameters
+            must be JSON serializable to support multiple task runners implementations.
+
+        :type function: str
+        """
+        assert isinstance(function, str)
+
+        try:
+            json.dumps(kwargs)
+        except TypeError:
+            raise TaskNotSerializableError("Task kwargs is not serializable.")
+
+        self.function = function
+        self.kwargs = kwargs
+
+
+class TaskRunner(object):
```

Comment:
Should probably get used to just leaving out the `(object)` from all new code we write, since it has no meaning in Python 3.

Reply:
Makes sense, we are used to write python2 compatible code 🤷

Reply:
Hopefully those days are over now :D

Reply:
many years doing this, give me some time :)